### DISCUSSION TITLE
VSCode / PlatformIO compile

### DISCUSSION THREAD

**Discussion by OK1FWG** [SQ9MDD/TTGO-T-Beam-LoRa-APRS#53] [General]
Any success story how to compile in PlatformIO? I am fighting with it a little bit. All other LoRa projects are not so difficult. :-)

**Comment by mattbk** [comment]
I managed to get through it. Have you seen this document? https://github.com/SQ9MDD/TTGO-T-Beam-LoRa-APRS/blob/master/INSTALL.md

I am on macOS, and after installing PlatformIO in VSCode, I got a popup about how compiling might not work because of some conflict. I spent a brief amount of time trying to fix the popup, but in the end I ignored it and things seem to work. 

Notably, I learned that you don't need to compile for all devices. This is a screenshot that hopefully illustrates what to click once the libraries have been installed, for one device:
<img width="342" alt="image" src="https://user-images.githubusercontent.com/3729038/132132013-dc46dd4c-9e1a-4c59-ac88-9b96574e3e2f.png">

This is not my normal development environment, but I was able to modify the code, compile, and flash two boards. If you run into any problems, I'm happy to try to help.

**Reply by OK1FWG** [reply]
Hi,
thanks for pointing me to the document. With some ideas from other project, I got it working. I am not using all libraries in general storage. So adding them as dependency to this project and some tweaking with includes, it compiles fine. Also the forward declaration of some functions was necessary, but it is okay. Thanks!
